pkg/config/config.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2024 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"crypto/tls"
+	"fmt"
+	"os"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	configapi "github.com/kubeflow/trainer/v2/pkg/apis/config/v1alpha1"
+)
+
+// fromFile loads configuration from a file.
+func fromFile(path string, scheme *runtime.Scheme, cfg *configapi.Configuration) error {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return fmt.Errorf("failed to read config file: %w", err)
+	}
+
+	codecs := serializer.NewCodecFactory(scheme, serializer.EnableStrict)
+
+	// Decode the configuration file into the Configuration object
+	if err := runtime.DecodeInto(codecs.UniversalDecoder(), content, cfg); err != nil {
+		return fmt.Errorf("failed to decode config file: %w", err)
+	}
+
+	return nil
+}
+
+// validate validates the configuration.
+func validate(cfg *configapi.Configuration) field.ErrorList {
```

Comment:
Do we want to move validation to the separate file, like in Kueue: https://github.com/kubernetes-sigs/kueue/blob/main/pkg/config/validation.go ?